### PR TITLE
Add packager for FIS specific ISO spec

### PR DESCRIPTION
**Pull request by ForageRob**
Need to set useHeader to False for FIS messaging.

I changed the lib/iso8583/packer/ans.js file so that it no longer works with the other packager classes (i.e. openWay.js and smartVista.js).

Added a new packer for x + n data that hasn't existed before.

### CODE REVIEW

File: lib/iso8583/lib/packer/xn.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+exports.help = function() {
+    return 'Fixed numeric value, zero padded, with leading C or D to indicate credit or debit';
+  };
+
+  exports.unpack = function(msg, packager) {
```
Comment (tuxtitlan) [inline lib/iso8583/lib/packer/xn.js:5]: nitpick: inconsistent indentation

File: lib/iso8583/lib/packager/fis.js [hunk 2]
```diff
@@ -0,0 +1,568 @@
...
+    '912': 'Timeout Waiting for Response',
+    '913': 'Duplicate Transaction Received',
+    '914': 'Could not Find the Original Transaction',
+    '915': 'Amount Being Reversed is Greater than Original, or no Amount Being Reversed',
+    '920': 'Pin Processing Error',
+    '923': 'Request in Progress'
+  }
+};
+
+exports.format = {
```
Comment (tuxtitlan) [inline lib/iso8583/lib/packager/fis.js:106]: This full list of data elements is awesome

File: lib/iso8583/lib/packager/fis.js [hunk 3]
```diff
@@ -0,0 +1,568 @@
...
+  expirationLimitSeconds: 60,
+
+  // The following errors should be treated as system or critical
+  sysErrors: [],
+
+  // The values of the following fields should be padded automatically
+  padFields: [],
+
+  // Reference of obligatory fields for different MTIs
+  obligatoryFields: {
```
Comment (tuxtitlan) [inline lib/iso8583/lib/packager/fis.js:24]: I'm wondering if all message types captured under `Message Formats` (https://docs.google.com/document/d/1KkAAtAFz7O1iqG8p0EKcBBmhr02VEg_2/edit#heading=h.1vsw3ci) should be included here

File: lib/iso8583/lib/packager/fis.js [hunk 4]
```diff
@@ -0,0 +1,568 @@
...
+    '01': 'Manual',
+    '06': 'Manual',
+    '02': 'Magnetic stripe read',
+    '05': 'Integrated circuit card read; CVV data reliable',
+    '07': 'Proximity transaction originated using ICC data rules',
+    '08': 'Magnetic stripe even though it is ICC capable',
+    '91': 'Proximity transaction originated using magnetic stripe data rules; CVV check is possible'
+  },
+
+  cardholderAuthMethods: {
```
Comment (tuxtitlan) [inline lib/iso8583/lib/packager/fis.js:41]: same here with regards to `PIN Entry Mode Codes (022-2) (Cardholder Verification Method)` (https://docs.google.com/document/d/1KkAAtAFz7O1iqG8p0EKcBBmhr02VEg_2/edit#heading=h.3k3xz3h)

File: lib/iso8583/lib/packager/fis.js [hunk 5]
```diff
@@ -0,0 +1,568 @@
...
+
+  // Reference of obligatory fields for different MTIs
+  obligatoryFields: {
+    '0800': [0, 1, 7, 11, 70],
+    '0810': [0, 1, 7, 11, 39, 70],
+    '0200': [0, 2, 3, 4, 7, 11, 12, 13, 18, 22, 32, 37, 41, 42, 43, 48, 49, 58],
+    '0210': [0, 2, 3, 4, 7, 11, 12, 13, 15, 18, 22, 32, 37, 39, 41, 42, 43, 48, 49, 58],
+  },
+
+  cardDataInputModes: {
```
Comment (tuxtitlan) [inline lib/iso8583/lib/packager/fis.js:31]: same here with regards to `PAN Entry Mode Codes (022-1) (Card Reading Method)` (https://docs.google.com/document/d/1KkAAtAFz7O1iqG8p0EKcBBmhr02VEg_2/edit#heading=h.2kz067v)

**Review by tuxtitlan — CHANGES_REQUESTED** [review]

File: lib/iso8583/lib/packager/fis.js [hunk 6]
```diff
@@ -0,0 +1,568 @@
...
+  cardholderAuthMethods: {
+    0: 'Not Authenticated',
+    1: 'PIN',
+    2: 'Signature Based',
+    6: 'Signature Based',
+    9: 'Terminal accept Off-line PINs (for EMV cards)'
+  },
+
+  // System error messages reference by code
+  errors: {
```
[the remark this answers is not in the document]
Comment (ForageRob) [reply lib/iso8583/lib/packager/fis.js:57]: Agreed that we need to modify this logic. I changed `parseInt(code)` to `code.trim()`. Might not be robust enough, but we can revisit this issue later once we can actually test end-to-end.

File: lib/iso8583/lib/packager/fis.js [hunk 7]
```diff
@@ -0,0 +1,569 @@
...
+  // This method is triggered for every packet before it is being packed
+  beforePack: function(data) {
+    return data;
+  }
+};
+
+exports.generators = {
+  getHeader: function(lengthVal) {
+    var result = Buffer.from('');
+    var headerLength = exports.processors.getHeaderLength();
```
Comment (tuxtitlan) [inline lib/iso8583/lib/packager/fis.js:551]: declaration for `getHeaderLength` seems to be missing

Comment (tuxtitlan) [reply lib/iso8583/lib/packager/fis.js:551]: will be resolved in other PR

**Review by tuxtitlan — APPROVED** [review]

**Pull request merged**